src/config/user.js

```diff
@@ -125,6 +125,54 @@ const defaults = {
   }
 }
 
+/**
+ * @typedef {(filepath: string) => Promise<Options | {}>} Loader
+ */
+
+/**
+ *
+ * @param {{default?: Options} | Options} mod
+ * @returns {Options | {}}
+ */
+const handleConfigImport = (mod) => {
+  /**
+   * @type {*}
+   */
+  const modWithDefaultExport = mod
+  if (modWithDefaultExport.default != null) {
+    return modWithDefaultExport.default
+  }
```

Comment:
jsdoc  + ts-check pragma doesn't handle union types well and didn't want to let me check for the default property, nor even use it in the true clause block.